src/main/java/net/jodah/failsafe/internal/util/CircularBitSet.java

```diff
@@ -22,8 +22,24 @@ public CircularBitSet(int size, CircularBitSet oldBitSet) {
 
     // Initialize from oldBitSet
     if (oldBitSet != null) {
-      for (int i = 0; i < size && i < oldBitSet.occupiedBits; i++)
+      initializeFromOldBits(oldBitSet);
+    }
+  }
+
+  /**
+   * Initialize bitSet from oldBitSet with correct order
+   */
+  private void initializeFromOldBits(CircularBitSet oldBitSet) {
```

Comment:
I think we can even simplify this a bit, something like:

```
  private void initializeFromOldBits(CircularBitSet oldBitSet) {
    int startIndex = oldBitSet.occupiedBits <= size ? 0 : oldBitSet.occupiedBits - size;
    for (int i = startIndex; i < oldBitSet.occupiedBits; i++)
      setNext(oldBitSet.bitSet.get(i));
  }
```

What do you think?